File: MojStomatolog/MojStomatolog.Models/Core/OrderItem.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace MojStomatolog.Models.Core
{
    public class OrderItem
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public int OrderId { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }

        [ForeignKey("ProductId")]
        public Product Product { get; set; } = null!;

        [ForeignKey("OrderId")]
        [JsonIgnore]
        public Order Order { get; set; } = null!;
    }
}

File: MojStomatolog/UI/mojstomatolog_desktop/lib/models/order_item.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mojstomatolog_desktop/models/product.dart';

part 'order_item.g.dart';

@JsonSerializable()
class OrderItem {
  int? id;
  int? productId;
  int? orderId;
  int? quantity;
  double? price;
  Product? product;

  OrderItem();

  factory OrderItem.fromJson(Map<String, dynamic> json) =>
      _$OrderItemFromJson(json);
  Map<String, dynamic> toJson() => _$OrderItemToJson(this);
}
